src/electron/go_vpn_tunnel.ts

```diff
@@ -150,8 +139,6 @@ export class GoVpnTunnel implements VpnTunnel {
     }
 
     console.log('restarting tun2socks after resume');
-
-    this.tun2socks.onExit = this.tun2socksExitListener;
     this.tun2socks.start(this.isUdpEnabled);
 
     // Check if UDP support has changed; if so, silently restart.
```

Comment:
Could you rename `retestUdp` to `updateUdpSupport` or something like that. I find the restest name misleading because it makes me think it's just testing something and not changing any state.